src/backend/direct.rs

```diff
@@ -614,9 +628,12 @@ impl crate::Context for Context {
         };
         let global = &self.0;
         wgc::gfx_select!(
-            *device => global.device_create_shader_module(*device, desc, PhantomData)
+            device.id => global.device_create_shader_module(device.id, desc, PhantomData)
```

Comment:
we could probably figure out a slightly shorter way of doing this, although I'm not sure if we should (given the complexity). I.e. this could be
```rust
device.id => device.create_wrap(
    &self.0,
    |g| g.device_create_shader_module(device.id, desc, PhantomData),
    |g| g.shader_module_error(PhantomData),
)
```
It slightly feels like fighting against the `gfx_select` macro limitations.
It's up to you to decide if the current code needs to stay, or something like this used instead.

Reply:
Hmm. It is certainly tempting to try to hide some of this. The closure in the unwrap is certainly a little awkward. But it this also has a little more implicit feel.

I started thinking, could I implement some `ErrorId` trait for the various id types, so the unwrap could just call `T::error_id()`, and not have the closure? But more work and also has a little implicit feel, especially as each trait impl probably has a single place of use.

I think I'm gonna leave it as is, unless something clearly better comes to mind